### PR TITLE
FEAT: add a list of bad words and reject a question that contains bad word

### LINKED ISSUE #66: A static block list words
**Is your feature request related to a problem? Please describe.**
I want to have blocklist words that used to block some bad words.

**Describe the solution you'd like**
Create an Array or Map of words. Split the question by the space character, make sure it's not contained the bad words.

The check process should be done in the very beginning before doing any Database select to reduce unnecessary traffic to DB. Early return if it's not passing the check.

### PR DESCRIPTION
**Pull request by madjiebimaa**
Closes #66 

## Description
Find a list of Indonesia bad words provided on the internet, format it as an Array, and use that Array to filter a question as a request from the Client.

## Ask
- Do I need to add question validation in the Client? because even though the server sent a 400 status code, React Query still runs the `onSuccess` condition. Based on [this](https://react-query.tanstack.com/guides/query-functions#usage-with-fetch-and-other-clients-that-do-not-throw-by-default) we need to handle errors manually if we use `fetch`.

### CODE REVIEW

**Comment by mazipan** [comment]
Use `httpClient` utils @madjiebimaa 
It will throw error on non-200 response

File: src/app/api/question/submit/route.ts [hunk 1]
```diff
@@ -76,6 +77,19 @@ async function sendQuestion(
 export async function POST(request: NextRequest) {
   const res = await request.json()
 
+  const isQuestionContainsBadWord =
+    res.question
+      .split(' ')
+      .filter((word: string) => BAD_WORDS.includes(word.toLowerCase())).length >
```
Comment (mazipan) [inline src/app/api/question/submit/route.ts]: Prefer using key-value data stucture, to reduce the time complexity.

https://www.tech-hour.com/javascript-performance-and-optimization

**Comment by mazipan** [comment]
What is the error message showing to the user when we face the bad words error?

We need to differentiate the error message, so user can understand that the question is containing bad words.

**Review by mazipan — APPROVED** [review]

**Comment by madjiebimaa** [comment]
> What the error showing on the user when we hit the bad words error? We need to differentiate the error message. So user can understand that the question is containing bad words.

The error message is still the same as the previous version. Okay mas, I will change it.

**Comment by mazipan** [comment]
I will merge this PR first, we can create a separate PR for fixing the error message

**Pull request merged**